site/src/sphinx/client-custom-http-headers.rst

```diff
@@ -78,8 +78,8 @@ If you want more freedom on how you manipulate the request headers, use a decora
 
     // Add a decorator that inserts the custom header.
     cb.decorator((delegate, ctx, req) -> { // See DecoratingHttpClientFunction and DecoratingRpcClientFunction.
-        req.headers().set(AUTHORIZATION, credential);
-        return delegate.execute(ctx, req);
+        RequestHeaders req0 = req.headers().set(AUTHORIZATION, credential).toBuilder().build();
```

Comment:
How about naming it `authorized` instead of `req0`?

Reply:
Also I think this is `req.headers().toBuilder().set(...).build();`

Reply:
[`RequestHeader`](https://line.github.io/armeria/apidocs/com/linecorp/armeria/common/RequestHeaders.html) is immutable. It doesn't have `set(...)` methods. We need to call `toBuilder()` to modify and set another attribute to it.  😀

Reply:
And we need to update after https://github.com/line/armeria/pull/2283 is merged.

Reply:
> RequestHeader is immutable. It doesn't have set(...) methods. We need to call toBuilder() to modify and set another attribute to it. 😀

Oh,, I see. Thanks for explanation!

Reply:
> And we need to update after #2283 is merged.

How do I deal with it?

It seems currently in `master` branch?

Reply:
I checked #2283, I notice that I need to use `HttpRequest.withHeaders` instead of `HttpRequest.of` ?

Reply:
Yes!, and don't forget to call `ctx.updateRequest`